ports/esp32/mpthreadport.c

```diff
@@ -116,11 +120,24 @@ void mp_thread_start(void) {
 static void *(*ext_thread_entry)(void *) = NULL;
 
 static void freertos_entry(void *arg) {
+    // Run the Python code.
     if (ext_thread_entry) {
         ext_thread_entry(arg);
     }
+
+    // Remove the thread from the linked-list of active threads.
+    mp_thread_mutex_lock(&thread_mutex, 1);
+    for (mp_thread_t **th = &thread; *th != NULL; th = &(*th)->next) {
+        if ((*th)->id == xTaskGetCurrentTaskHandle()) {
+            *th = (*th)->next;
+        }
+    }
+    mp_thread_mutex_unlock(&thread_mutex);
+
+    // Delete this FreeRTOS task.
     vTaskDelete(NULL);
-    for (;;) {;
+    for (;;) {
+        vTaskSuspend(NULL);
```

Comment:
Should be safe to remove this infinite loop.

Reply:
Now removed, and comment updated.